src/Tribe/Values/Abstract_Value.php

```diff
@@ -0,0 +1,423 @@
+	}
+
+	/**
+	 * Takes the value string and a list of non-digit characters and removes any of those characters. If the character
+	 * is found to be a decimal separator, normalize it to a dot, so the number translates to a float.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $value      the value being normalized
+	 * @param string[] $non_digits a list of non-digit characters present in $value
+	 * @param string   $separator  a default separator to use when splitting the string
+	 *
+	 * @return string[]
+	 */
+	private function remove_non_digits( $value, $non_digits, $separator = '>>>' ) {
+
+		$tokens = array_unique( $non_digits[0] );
+
+		foreach ( $tokens as $token ) {
+			if ( $this->is_decimal_separator( $token, $value ) ) {
+				$separator = $token;
+				continue;
+			}
+
+			$value = str_replace( $token, '', $value );
+		}
+
+		return explode( $separator, $value );
+	}
+
+	/**
+	 * Re-assemble the normalized value to store.
+	 *
+	 * @since TBD
+	 *
+	 * @param int[] $pieces the normalized value split in an array.
+	 *
+	 * @return float
+	 */
+	private function assemble_normalized_value( $pieces ) {
+
+		// If the initial amount did not have decimals specified, $pieces will be an array of a single
+		// numeric value, so we just return it as a float.
+		if ( 1 === count( $pieces ) && is_numeric( reset( $pieces ) ) ) {
+			return (float) reset( $pieces );
+		}
+
+		$decimal = array_pop( $pieces );
+
+		return (float) implode( '', array_merge( $pieces, [ '.', $decimal ] ) );
+	}
+
+	/**
+	 * Private setter for the initial value the object was created with. This value cannot be changed during the object
+	 * lifecycle.
+	 *
+	 * @since TBD
+	 *
+	 * To set a new value discard the original object and create a new one.
+	 */
+	private function set_initial_representation( $amount ) {
+		if ( empty( $this->initial_value ) ) {
+			$this->initial_value = $amount;
+		}
+	}
+
+	/**
+	 * Private setter for the normalized amount extracted from the initial value.
+	 *
+	 * @since TBD
+	 *
+	 * To set a new value use the public setter `$obj->set_value( $amount )`
+	 */
+	private function set_normalized_amount( $amount ) {
+
+		$normalized_value = $this->normalize( $amount );
+
+		/**
+		 * Filter the value to be set as $normalized_amount for a specific implementation.
+		 *
+		 * @since TBD
+		 *
+		 * @param float $normalized_value the normalized value
+		 * @param Abstract_Value the object instance
+		 *
+		 * @return float
+		 */
+		$normalized_value = (float) apply_filters( "tec_common_{$this->get_class_name()}_value_normalized", $normalized_value, $this );
+
+		/**
+		 * Filter the value to be set as $normalized_amount for all implementations.
+		 *
+		 * @since TBD
+		 *
+		 * @param float $normalized_value the normalized value
+		 * @param Abstract_Value the object instance
+		 *
+		 * @return float
+		 */
+		$normalized_value = (float) apply_filters( "tec_common_value_normalized", $normalized_value, $this );
```

Comment:
You might have a reason for doing it this way, but it would seem more consistent to put these within the `get_normalized_value()` method.

Reply:
There is a reason for this. In short: every single operation that takes place using any class that extends `Abstract_Value` relies on the response to `get_normalized_value()` being uniform. This makes it risky to add filters to it, because you can have filters return different values for the same object, and that has the potential of screwing up every calculation and possibly every formatting function too. Adding the filters before setting the parameter value is how we go around that problem. You can still change what's going to be in the parameter, but we make sure the value you choose to return will be the same for every calculation.